preview/Win7Msix/Win7MSIXInstaller/ValidateRequest.cpp

```diff
@@ -0,0 +1,88 @@
+#include <windows.h>
+
+#include <shlobj_core.h>
+#include <CommCtrl.h>
+
+#include "FilePaths.hpp"
+#include "ValidateRequest.hpp"
+#include "GeneralUtil.hpp"
+#include <TraceLoggingProvider.h>
+#include <VersionHelpers.h>
+
+const PCWSTR ValidateRequest::HandlerName = L"ValidateRequest";
+
+HRESULT ValidateRequest::ExecuteForAddRequest()
+{
+    RETURN_IF_FAILED(ParseDependencyInfoFromPackage());
+    RETURN_IF_FAILED(MatchTDFAndVersionToOS());
+    return S_OK;
+}
+
+HRESULT ValidateRequest::ParseDependencyInfoFromPackage()
+{
+    PackageInfo* packageInfo = m_msixRequest->GetPackageInfo();
+
+    ComPtr<IMsixDocumentElement> domElement;
+    RETURN_IF_FAILED(packageInfo->GetManifestReader()->QueryInterface(UuidOfImpl<IMsixDocumentElement>::iid, reinterpret_cast<void**>(&domElement)));
+
+    ComPtr<IMsixElement> element;
+    RETURN_IF_FAILED(domElement->GetDocumentElement(&element));
+
+    // Obtain the TargetDeviceFamily tag info
+    ComPtr<IMsixElementEnumerator> dependencyEnum;
+    RETURN_IF_FAILED(element->GetElements(
+        L"/*[local-name()='Package']/*[local-name()='Dependencies']/*[local-name()='TargetDeviceFamily']",
+        &dependencyEnum));
+
+    BOOL hc = FALSE;
+    RETURN_IF_FAILED(dependencyEnum->GetHasCurrent(&hc));
+    if (!hc)
+    {
+        TraceLoggingWrite(g_MsixTraceLoggingProvider,
+            "No Target device family Found",
+            TraceLoggingLevel(WINEVENT_LEVEL_ERROR));
+        return E_NOT_SET;

```

Comment:
>E_NOT_SET [](start = 15, length = 9)

i think any failures coming from rejected packages due to this should have a more unique error. ERROR_INSTALL_REJECTED seems like it has the right string associated with it:  The app that you are trying to run is not supported on this version of Windows.

Reply:
Updated.